app/models/allyship.rb

```diff
@@ -57,4 +58,21 @@ def different_users
     errors.add(:user_id, 'user_id is nil') if user_id.nil?
     errors.add(:ally_id, 'ally_id is nil') if ally_id.nil?
   end
+
+  private
+
+  def remove_activities_between_users
+    user_id = self.user_id
+    ally_id = self.ally_id
+
+    Notification.for_ally(user_id, ally_id).or(
```

Comment:
We could move this into its own helper

Reply:
Do you think moving this part to concen would be more appropriate?

Reply:
We could move it into a concern, but will it be reused again? If not I think it's fine just sticking that logic for destroying Notifications into a private method in the model (same goes for removing allies from viewers lists).

Reply:
I was thinking you should update it to this:

```
private

def remove_ally_notifications
   user_id = self.user_id
   ally_id = self.ally_id
   Notification.for_ally(user_id, ally_id).or(
      Notification.for_ally(ally_id, user_id)
    ).destroy_all
end

def remove_ally_viewers
   user_id = self.user_id
   ally_id = self.ally_id
   [Moment, Strategy].each do |viewed_class|
      viewed_class.destroy_viewer(user_id, ally_id)
      viewed_class.destroy_viewer(ally_id, user_id)
    end
end

def remove_activities_between_users
    remove_ally_notifications
    remove_ally_viewers
  end
```

Reply:
Thank you, I'll update it right now!